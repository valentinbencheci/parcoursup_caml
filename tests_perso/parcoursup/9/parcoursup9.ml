(* ###################### - DESCRIPTION TEST - ###################### *)
(* On ajoute 2 fois le meme candidat et on regarde si on n'aura pas 
des dublicates. En plus on ajoute une formation pour laquelle il n'y a 
pas des candidatures. *)
(* ################################################################## *)

open Test_parcoursup_utils

let session = nouvelle_session ()

let () =
  ajoute_candidat session ~nom_candidat:"Adam";
  ajoute_candidat session ~nom_candidat:"Leo";
  ajoute_candidat session ~nom_candidat:"Adam";

  ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:1;
  ajoute_formation session ~nom_formation:"Université de Toulon" ~capacite:1;

  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session
    ~rang_repondeur:(Some 1)
    ~nom_candidat:"Leo"
    ~nom_formation:"Université Côte d'Azur";

  ajoute_commission session
    ~nom_formation:"Université Côte d'Azur"
    ["Adam";"Leo"];
  ajoute_commission session
    ~nom_formation:"Université de Toulon"
    ["Adam";"Leo"];

  reunit_commissions session;

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Leo";
  affiche_propositions_en_attente session "Leo";

  renonce session
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Leo";
  affiche_propositions_en_attente session "Leo";