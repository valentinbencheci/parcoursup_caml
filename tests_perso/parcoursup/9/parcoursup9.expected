nouvelle session
ajoute candidat Adam
ajoute candidat Leo
ajoute candidat Adam
ajoute formation Université Côte d'Azur (1 places)
ajoute formation Université de Toulon (1 places)
ajoute voeu de Adam pour Université Côte d'Azur (rang=1)
ajoute voeu de Leo pour Université Côte d'Azur (rang=1)
commission de classement de Université Côte d'Azur : [Adam; Leo]
commission de classement de Université de Toulon : [Adam; Leo]
réunit commissions
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université Côte d'Azur}
Voeux en attente de Leo : {Université Côte d'Azur(0)}
Propositions en attente de Leo : {}
Adam renonce à Université Côte d'Azur
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {}
Voeux en attente de Leo : {}
Propositions en attente de Leo : {Université Côte d'Azur}
